doc/examples/segmentation/plot_euler_number.py

```diff
@@ -0,0 +1,140 @@
+in 2D and 3D objects.
+
+For 2D objects, the Euler number is the number of objects minus the number of
+holes. Notice that if an object is 8-connected, its complementary set is
+4-connected, and conversely. It is also possible to compute the number of
+objects using :func:`skimage.measure.label`, and to deduce the number of holes
+from the difference between the two numbers.
+
+For 3D objects, the Euler number is obtained as the number of objects plus the
+number of holes, minus the number of tunnels, or loops. If an
+object is 26-connected, its complementary set is 6-connected, and conversely.
+The voxels are actually represented with blue transparent surfaces.
+Inner porosities are represented in red.
+
+"""
+from mpl_toolkits.mplot3d import Axes3D
+from skimage.measure import euler_number, label
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+# Sample image.
+SAMPLE = np.array(
+    [[0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0],
+     [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+     [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+     [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+     [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+     [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+     [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+     [1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0],
+     [0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1],
+     [0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1]]
+)
+SAMPLE = np.pad(SAMPLE, 1, mode='constant')
+
+fig, ax = plt.subplots()
+ax.imshow(SAMPLE, cmap=plt.cm.gray)
+ax.axis('off')
+e4 = euler_number(SAMPLE, connectivity=1)
+object_nb_4 = label(SAMPLE, connectivity=1).max()
+holes_nb_4 = object_nb_4 - e4
+e8 = euler_number(SAMPLE, connectivity=2)
+object_nb_8 = label(SAMPLE, connectivity=2).max()
+holes_nb_8 = object_nb_8 - e8
+ax.set_title('Euler number for N4: {} ({} objects, {} holes), \n for N8: {} ({} objects, {} holes)'.format(e4, object_nb_4, holes_nb_4, e8, object_nb_8, holes_nb_8))
+plt.show()
+
+######################################################################
+# 3-D objects
+# ===========
+#
+# In this example, a 3-D cube is generated, then holes and
+# tunnels are added. Euler number is evaluated with 6 and 26 neighborhood
+# configuration. This code is inpired by
+# https://matplotlib.org/devdocs/gallery/mplot3d/voxels_numpy_logo.html
+
+
+def make_ax(grid=False):
+    fig = plt.figure()
+    ax = fig.gca(projection='3d')
+    ax.grid(grid)
+    ax.set_axis_off()
+    return ax
+
+
+def explode(data):
+    """visualization to separate voxels
+    """
+    size = np.array(data.shape) * 2
+    data_e = np.zeros(size - 1, dtype=data.dtype)
+    data_e[::2, ::2, ::2] = data
+    return data_e
+
+# shrink the gaps between voxels
+
+
+def expand_coordinates(indices):
+    x, y, z = indices
+    x[1::2, :, :] += 1
+    y[:, 1::2, :] += 1
+    z[:, :, 1::2] += 1
+    return x, y, z
```

Comment:
![Figure_2](https://user-images.githubusercontent.com/263366/93184334-656d9400-f73c-11ea-83e3-52426195bc66.png)

There are a series of tricks in this PR so that voxels appear properly. If you just use an array of indices here is how it looks like. I tried to add some more comments, I agree it's a bit cryptic (remember that it's a PR I took over :-))